huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_dedicated_instance_test.go

```diff
@@ -0,0 +1,187 @@
+					resource.TestCheckResourceAttr(resourceName, "name", name+"_updated"),
+					resource.TestCheckResourceAttr(resourceName, "available_zone", "cn-north-4b"),
+					resource.TestCheckResourceAttr(resourceName, "cpu_architecture", "x86"),
+					resource.TestCheckResourceAttr(resourceName, "specification_code", "waf.instance.professional"),
+					resource.TestCheckResourceAttr(resourceName, "security_group.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "run_status", "1"),
+					resource.TestCheckResourceAttr(resourceName, "access_status", "0"),
+					resource.TestCheckResourceAttr(resourceName, "upgradable", "0"),
+					resource.TestCheckResourceAttrSet(resourceName, "server_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "service_ip"),
+					resource.TestCheckResourceAttrSet(resourceName, "subnet_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "vpc_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "cpu_flavor"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{},
+			},
+		},
+	})
+}
+
+func testAccCheckWafDedicatedInstanceV1Destroy(s *terraform.State) error {
+	config := acceptance.TestAccProvider.Meta().(*config.Config)
+	c, err := config.WafDedicatedV1Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return fmt.Errorf("error creating HuaweiCloud WAF dedicated client: %s", err)
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "huaweicloud_waf_dedicated_instance" {
+			continue
+		}
+		_, err := instances.GetInstance(c, rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Waf domain still exists")
+		}
+	}
+	return nil
+}
+
+func testAccCheckWafDedicatedInstanceV1Exists(n string, instance *instances.DedicatedInstance) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := acceptance.TestAccProvider.Meta().(*config.Config)
+		c, err := config.WafDedicatedV1Client(acceptance.HW_REGION_NAME)
+		if err != nil {
+			return fmt.Errorf("error creating HuaweiCloud WAF dedicated client: %s", err)
+		}
+
+		found, err := instances.GetInstance(c, rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if found.Id != rs.Primary.ID {
+			return fmt.Errorf("Waf dedicated instance not found")
+		}
+		*instance = *found
+		return nil
+	}
+}
+
+func baseDependResource(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_vpc" "vpc_1" {
+  name = "%s_waf"
+  cidr = "192.168.0.0/24"
+}
+
+resource "huaweicloud_vpc_subnet" "vpc_subnet_1" {
+  name          = "%s_waf"
+  cidr          = "192.168.0.0/24"
+  gateway_ip    = "192.168.0.1"
+  primary_dns   = "100.125.1.250"
+  secondary_dns = "100.125.129.250"
```

Comment:
you can use the default dns as the region is not fixed to cn-north-4.

Reply:
removed them